ecs.js

```diff
@@ -344,6 +344,25 @@ function _removeComponent (world, entity, componentName) {
     }
 }
 
+// purpose: by given filterId and component determine if component is referred in that filter.
+// arguments: filterId is a string in the form "component1,component2,...,componentN", component is a string
+function _hasComponent (filterId, component) {
+  return (filterId === component) ||
+         filterId.startsWith(`${component},`) ||
+         filterId.endsWith(`,${component}`) ||
+         filterId.includes(`,${component},`)
+}
+
+/* Seems provided algorithm for _hasComponent is pretty good. Alternatives:
```

Comment:
1-3 are good points but you don't need to include them in the PR :)

Reply:
Hi Mike! May I ask you why there is no need to include them? I have some opinion but really want to hear yours.